### PR TITLE
Readstream

### PR DESCRIPTION
**Pull request by perry-mitchell**
Readstream support

### CODE REVIEW

**Comment by perry-mitchell** [comment]
Encoding may be required before closing this.

File: source/index.js [hunk 1]
```diff
@@ -40,6 +47,25 @@ module.exports = function(webDAVEndpoint, username, password) {
...
+         */
+        createReadStream: function(filePath, options) {
+            var clientOptions = {};
+            if (options && options !== null) {
+                if (typeof options.headers === "object") {
+                    clientOptions.headers = options.headers;
+                }
+                if (typeof options.start === "number" && typeof options.end === "number") {
+                    clientOptions.range = { start: options.start, end: options.end };
+                }
```
Comment (piranna) [inline source/index.js]: I would allow to don't require to have defined both `start` and `end`, by default they could be zero and +Infinite respectively.

Comment (perry-mitchell) [inline source/index.js]: Hmm true. Don't recall where in the docs it said about the byte range start and end, but I guess we should mirror that. Unfortuantely [webdav-client has this requirement](https://github.com/perry-mitchell/webdav-client/blob/master/source/adapter/get.js#L78) also currently - It assumes both are there. Hmm..

Comment (perry-mitchell) [inline source/index.js]: I haven't yet found any docs on byte ranges allowing for only start. For instance: https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests

**Comment by ultimate-tester** [comment]
This is the page you should be looking at:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range

Especially this example:
**Range: bytes 200-1000, 2000-6576, 19000-**

You see the header Range: 19000- would indicate that you want from byte 19000 till the end. This is what you're looking for.

**Comment by perry-mitchell** [comment]
@ultimate-tester Good link, thanks. 

I don't think we'll support these however:

```
Range: <unit> <range-start>-<range-end>, <range-start>-<range-end>
Range: <unit> <range-start>-<range-end>, <range-start>-<range-end>, <range-start>-<range-end>
```

Seeing as the `fs.createReadStream` method only supports one range.

**Comment by piranna** [comment]
> Seeing as the fs.createReadStream method only supports one range.

`webdav-client` would support it, although `webdav-fs` don't make use of it...

**Comment by perry-mitchell** [comment]
Currently webdav-client supports a single range object, but adapting the API to also support an array like `[{start: 0, end: 99}, {start: 213, end: 554}]` would be quite simple. Imo I think we could move forward with what we have already.

**Comment by piranna** [comment]
> Imo I think we could move forward with what we have already.

Ok, I agree, let's add an issue for multiple ranges as remainder, though.

**Comment by perry-mitchell** [comment]
Added another commit updating webdav-client for ranges with `start` only.

**Pull request merged**